include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_radix_sort.h

```diff
@@ -514,21 +579,22 @@ __radix_sort_reorder_submit(_ExecutionPolicy&& __exec, ::std::size_t __segments,
 
                 _PeerHelper __peer_prefix_hlp(__self_item, __peer_temp);
 
-                // 1. create a private array for storing offset values
-                //    and add total offset and offset for compute unit for a certain radix state
-                _OffsetT __offset_arr[__radix_states];
                 const ::std::size_t __scan_size = __segments + 1;
-                _OffsetT __scanned_bin = 0;
-                __offset_arr[0] = __offset_rng[__wgroup_idx];
-                for (::std::uint32_t __radix_state_idx = 1; __radix_state_idx < __radix_states; ++__radix_state_idx)
+                for (::std::uint32_t __radix_state = __self_lidx; __radix_state < __radix_states; __radix_state += __sg_size)
```

Comment:
1) The changes are connected with the changes in __peer_prefix_helper class?
if not, may be it makes sense to do it in a separate PR?
2) And the second question  - how the changes (lines 583 - 596) influent to performance?

Reply:
For the first question, this PR is moving the loop over the radix states inside of the peer prefix class. This is because two of the peer prefix helpers (`atomic_ballot` and `fetch_or`) are now iterating over the radix bits instead of all of the radix states. I do not believe it makes sense to make a PR with just those changes since they are connected to the optimizations presented in this PR.

For the second question, the changes to store the offsets in local memory and perform a group scan in isolation creates a modest slowdown. However, storing the offsets in local memory and sharing them among the work-items is necessary for the changes to the `atomic_ballot` and `fetch_or` peer prefix helpers. Altogether, there is a performance speedup.